solutions/src/main/java/ru/sberbank/school/task10/MyScalableConcurrentThreadPool.java

```diff
@@ -0,0 +1,217 @@
+package ru.sberbank.school.task10;
+
+import lombok.NonNull;
+
+import java.util.concurrent.*;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.LongStream;
+
+/**
+ * Created by Mart
+ * 15.06.2019
+ **/
+public class MyScalableConcurrentThreadPool implements ThreadPool {
+    private final int sizeMin;
+    private final int sizeMax;
+    private final LinkedBlockingQueue<Long> freeIndexForNewWorker = new LinkedBlockingQueue<>();
+    private final ConcurrentHashMap<Long, ThreadWorker> poolMap = new ConcurrentHashMap<>();
+    private final LinkedBlockingQueue<FutureTask> tasks = new LinkedBlockingQueue<>();
+
+    private ReentrantLock tasksLock = new ReentrantLock();
+    private ReentrantReadWriteLock poolLock = new ReentrantReadWriteLock();
+    private ReentrantReadWriteLock.ReadLock poolReadLock = poolLock.readLock();
+    private ReentrantReadWriteLock.WriteLock poolWriteLock = poolLock.writeLock();
+
+    public MyScalableConcurrentThreadPool(int sizeMin, int sizeMax) {
+        if (sizeMin <= 0 || sizeMax <= 0 || sizeMax < sizeMin) {
+            throw new IllegalArgumentException("Incorrect input size");
+        }
+        this.sizeMin = sizeMin;
+        this.sizeMax = sizeMax;
+        fillPoolMap(sizeMin);
+    }
+
+    public MyScalableConcurrentThreadPool(int fixedSize) {
+        if (fixedSize <= 0) {
+            throw new IllegalArgumentException("Incorrect input size");
+        }
+        this.sizeMin = fixedSize;
+        this.sizeMax = fixedSize;
+        fillPoolMap(fixedSize);
+    }
+
+    private void fillPoolMap(long size) {
+        try {
+            poolWriteLock.lock();
+            LongStream.range(0L, size)
+                    .forEach(i -> poolMap.putIfAbsent(i, new ThreadWorker("ThreadWorker " + i, i)));
+        }finally {
```

Comment:
WhitespaceAround: &#39;}&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-34122392>*